certbot-ci/certbot_integration_tests/utils/pebble_artifacts.py

```diff
@@ -32,11 +33,16 @@ def fetch(workspace: str, http_01_port: int = DEFAULT_HTTP_01_PORT) -> Tuple[str
 
 
 def _fetch_asset(asset: str, assets_path: str) -> str:
-    platform = 'linux-amd64'
     base_url = 'https://github.com/letsencrypt/pebble/releases/download'
-    asset_path = os.path.join(assets_path, f'{asset}_{PEBBLE_VERSION}_{platform}')
+    os_type = platform.system().lower()  # this will be something like "darwin" or "linux"
+    architecture = platform.machine()
+    if architecture == 'x86_64':
+        architecture = 'amd64'
+    elif architecture == 'aarch64':
+        architecture = 'arm64'
```

Comment:
nit:

```suggestion
        architecture = 'arm64'
    else:
        raise SomeError()
```

or use a different variable rather than re-assigning and then assert `architecture` is set; would just prefer to make our future lives a bit easier

Reply:
i was going to just rely on the error message from failing to fetch the resource from GitHub a few lines down, but we could make that message more useful. since we could also have problems with the `os_type = platform.system().lower()` line, how about something like my recent commit?

with this approach, a failed run looks like:
```
$ tox -e integration
integration: commands[0]> python -m pytest certbot-ci/certbot_integration_tests --cov=acme --cov=certbot --cov=certbot_nginx --cov-report= --cov-config=certbot-ci/certbot_integration_tests/.coveragerc
ACME xdist config:
{'directory_url': 'https://localhost:14000/dir', 'challtestsrv_url': 'http://localhost:8055', 'http_port': {'gw0': 5200, 'gw1': 5201, 'gw2': 5202, 'gw3': 5203, 'gw4': 5204, 'gw5': 5205, 'gw6': 5206, 'gw7': 5207}, 'https_port': {'gw0': 5100, 'gw1': 5101, 'gw2': 5102, 'gw3': 5103, 'gw4': 5104, 'gw5': 5105, 'gw6': 5106, 'gw7': 5107}, 'other_port': {'gw0': 5300, 'gw1': 5301, 'gw2': 5302, 'gw3': 5303, 'gw4': 5304, 'gw5': 5305, 'gw6': 5306, 'gw7': 5307}}
=> Configuring the HTTP proxy on port 5002...
=> Finished configuring the HTTP proxy.
=> Starting pebble instance deployment...
=> Tear down the test infrastructure...
=> Test infrastructure stopped and cleaned up.
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/pebble_artifacts.py", line 50, in _fetch_asset
INTERNALERROR>     response.raise_for_status()
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/requests/models.py", line 1024, in raise_for_status
INTERNALERROR>     raise HTTPError(http_error_msg, response=self)
INTERNALERROR> requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://github.com/letsencrypt/pebble/releases/download/v2.7.0/pebble-imaginary-unicorn.zip
INTERNALERROR> 
INTERNALERROR> During handling of the above exception, another exception occurred:
INTERNALERROR> 
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/_pytest/main.py", line 279, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/_pytest/config/__init__.py", line 1118, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/pluggy/_hooks.py", line 535, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self._hookimpls.copy(), kwargs, False)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/.tox/integration/lib/python3.9/site-packages/pluggy/_callers.py", line 103, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/conftest.py", line 35, in pytest_configure
INTERNALERROR>     _setup_primary_node(config)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/conftest.py", line 100, in _setup_primary_node
INTERNALERROR>     acme_server.start()
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/acme_server.py", line 76, in start
INTERNALERROR>     raise e
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/acme_server.py", line 73, in start
INTERNALERROR>     self._prepare_pebble_server()
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/acme_server.py", line 124, in _prepare_pebble_server
INTERNALERROR>     pebble_artifacts_rv = pebble_artifacts.fetch(self._workspace, self._http_01_port)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/pebble_artifacts.py", line 28, in fetch
INTERNALERROR>     pebble_path = _fetch_asset('pebble', assets_path)
INTERNALERROR>   File "/Users/bmw/Development/github.com/certbot/certbot/certbot-ci/certbot_integration_tests/utils/pebble_artifacts.py", line 52, in _fetch_asset
INTERNALERROR>     raise ValueError(f'Unable to obtain pebble artifact for {os_type} {architecture}. Is '
INTERNALERROR> ValueError: Unable to obtain pebble artifact for imaginary unicorn. Is your platform supported?
=> Tear down the test infrastructure...
=> Test infrastructure stopped and cleaned up.
integration: exit 3 (0.42 seconds) /Users/bmw/Development/github.com/certbot/certbot> python -m pytest certbot-ci/certbot_integration_tests --cov=acme --cov=certbot --cov=certbot_nginx --cov-report= --cov-config=certbot-ci/certbot_integration_tests/.coveragerc pid=66358
  integration: FAIL code 3 (0.47=setup[0.05]+cmd[0.42] seconds)
  evaluation failed :( (0.56 seconds)
```

Reply:
lol beautiful error message. My main issue is less about notifying the user and more about having code that uses a) the response from the system and b) what's available on github that's not controlled by us. Both of these are uncomfortable and could cause problems if they change under us, and that's mitigated that by being explicit about what is allowed/expected. In practice it's, like, probably fine! But that's the sort of assumption that leads to future errors. I'm less worried about what happens if you use something that's not there than about what happens if something becomes available but we didn't expect it to be. I know that's unlikely here since we're the only callers, but I'd prefer to be strict about it just in case.

Reply:
i see. while i understand your point, i personally kind of like the permissive approach of just trying things that might work for dev tools like this. adding hardcoded restrictions makes things easier for future devs if they'd have broken and harder if they otherwise wouldn't

i'm fine going your way on this tradeoff though. how about something like my new commits?

Reply:
if we didn't have a translation step in there I'd be more willing to be permissive, but since we're taking strings and replacing them with other strings that rings my danger bell. thank you for doing this!